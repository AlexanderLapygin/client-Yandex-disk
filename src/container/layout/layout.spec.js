import React from 'react';
import { shallow } from 'enzyme';
import { Layout } from './Layout';

describe('Layout container', () => {

  const props = { // описываем props
    token: '',
    data: {
      item: [
        {
          name: 'folder1',
          resource_id: '12321434123',
          type: 'dir',
        },
        {
          name: 'folder2',
          resource_id: '1233244123',
          type: 'dir',
        },
      ],
    },
    currentPath: '',
    location: '/',
    fetchResourcesRequest: () => {
    },
  };

  describe('Layout container', () => {

    it('render initial', () => {
      const LayoutContainer = shallow(<Layout {...props} key='12732834'/>);

      expect(LayoutContainer.find('li')).toHaveLength(1);
    });
  });
});
