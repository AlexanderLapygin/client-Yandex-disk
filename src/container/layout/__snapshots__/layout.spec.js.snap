// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout container Layout container snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout
    currentPath=""
    data={
      Object {
        "items": Array [
          Object {
            "name": "folder1",
            "resource_id": "12321434123",
            "type": "dir",
          },
          Object {
            "name": "folder2",
            "resource_id": "1233244123",
            "type": "dir",
          },
        ],
      }
    }
    fetchInfoDiskRequest={
      [MockFunction] {
        "calls": Array [
          Array [
            "awdfq12ew",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    fetchResourcesRequest={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    info={
      Object {
        "display_name": "TestName",
      }
    }
    location="/"
    token="awdfq12ew"
    type="dir"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <header
        className="App-header bd-highlight"
      >
        <ul
          className="list-group list-resourse"
        >
          <withRouter(Connect(ListItem))
            item={
              Object {
                "name": "folder1",
                "resource_id": "12321434123",
                "type": "dir",
              }
            }
          />
          <withRouter(Connect(ListItem))
            item={
              Object {
                "name": "folder2",
                "resource_id": "1233244123",
                "type": "dir",
              }
            }
          />
        </ul>
      </header>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="list-group list-resourse"
        >
          <withRouter(Connect(ListItem))
            item={
              Object {
                "name": "folder1",
                "resource_id": "12321434123",
                "type": "dir",
              }
            }
          />
          <withRouter(Connect(ListItem))
            item={
              Object {
                "name": "folder2",
                "resource_id": "1233244123",
                "type": "dir",
              }
            }
          />
        </ul>,
        "className": "App-header bd-highlight",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            Array [
              <withRouter(Connect(ListItem))
                item={
                  Object {
                    "name": "folder1",
                    "resource_id": "12321434123",
                    "type": "dir",
                  }
                }
              />,
              <withRouter(Connect(ListItem))
                item={
                  Object {
                    "name": "folder2",
                    "resource_id": "1233244123",
                    "type": "dir",
                  }
                }
              />,
            ],
          ],
          "className": "list-group list-resourse",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": "12321434123",
            "nodeType": "function",
            "props": Object {
              "item": Object {
                "name": "folder1",
                "resource_id": "12321434123",
                "type": "dir",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1233244123",
            "nodeType": "function",
            "props": Object {
              "item": Object {
                "name": "folder2",
                "resource_id": "1233244123",
                "type": "dir",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "header",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <header
          className="App-header bd-highlight"
        >
          <ul
            className="list-group list-resourse"
          >
            <withRouter(Connect(ListItem))
              item={
                Object {
                  "name": "folder1",
                  "resource_id": "12321434123",
                  "type": "dir",
                }
              }
            />
            <withRouter(Connect(ListItem))
              item={
                Object {
                  "name": "folder2",
                  "resource_id": "1233244123",
                  "type": "dir",
                }
              }
            />
          </ul>
        </header>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="list-group list-resourse"
          >
            <withRouter(Connect(ListItem))
              item={
                Object {
                  "name": "folder1",
                  "resource_id": "12321434123",
                  "type": "dir",
                }
              }
            />
            <withRouter(Connect(ListItem))
              item={
                Object {
                  "name": "folder2",
                  "resource_id": "1233244123",
                  "type": "dir",
                }
              }
            />
          </ul>,
          "className": "App-header bd-highlight",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              Array [
                <withRouter(Connect(ListItem))
                  item={
                    Object {
                      "name": "folder1",
                      "resource_id": "12321434123",
                      "type": "dir",
                    }
                  }
                />,
                <withRouter(Connect(ListItem))
                  item={
                    Object {
                      "name": "folder2",
                      "resource_id": "1233244123",
                      "type": "dir",
                    }
                  }
                />,
              ],
            ],
            "className": "list-group list-resourse",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": "12321434123",
              "nodeType": "function",
              "props": Object {
                "item": Object {
                  "name": "folder1",
                  "resource_id": "12321434123",
                  "type": "dir",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1233244123",
              "nodeType": "function",
              "props": Object {
                "item": Object {
                  "name": "folder2",
                  "resource_id": "1233244123",
                  "type": "dir",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
        "type": "header",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
